sdks/python/src/opik/api_objects/rest_stream_parser.py

```diff
@@ -1,38 +1,64 @@
 import json
 import logging
-from typing import Iterable, Type, List, Optional, TypeVar
+from typing import Iterable, Set, Type, List, Optional, TypeVar, Protocol
 
 
 LOGGER = logging.getLogger(__name__)
 
 
-T = TypeVar("T")
+class _HasID(Protocol):
+    id: str
 
 
-def read_and_parse_stream(stream: Iterable[bytes], item_class: Type[T]) -> List[T]:
+T = TypeVar("T", bound=_HasID)
+
+
+def read_and_parse_stream(
+    stream: Iterable[bytes],
+    item_class: Type[T],
+    nb_samples: Optional[int] = None,
+    item_ids: Optional[Set[str]] = None,
```

Comment:
Let's move item_ids out from this function but keep the nb_samples.
item_ids make the API a bit confusing, lets make dataset.py responsible for filtering out the items (I also added more comments in dataset.py related to that)